boards/shared/Inc/App/App_SharedError.h

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+
+struct Error;
+struct ErrorList;
+struct ErrorBoardList;
+
+enum ErrorBoard
+{
+    BMS,
+    DCM,
+    DIM,
+    FSM,
+    PDM,
+    NUM_ERROR_BOARDS,
+};
+
+struct ErrorBoardList
+{
+    // Number of valid entries in "boards"
+    uint32_t num_boards;
+
+    // Storage large enough to store every board
+    enum ErrorBoard boards[NUM_ERROR_BOARDS];
+};
+
+enum ErrorId
+{
+    BMS_NON_CRIT,
```

Comment:
For now, I still stuck with making up just enough fake errors for writing unit tests. I will add in real errors from the DBC later.